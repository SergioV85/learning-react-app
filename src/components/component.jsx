import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}