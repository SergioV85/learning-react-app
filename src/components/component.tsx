import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}