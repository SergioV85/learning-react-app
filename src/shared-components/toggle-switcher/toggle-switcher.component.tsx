import * as React from 'react';

interface IToggleButton {
  icon?: string;
  label?: string;
  value: string;
}

interface IToggleSwitcherProps {
  buttonViewClass: string;
  buttons: IToggleButton[];
  currentValue: boolean | number | string;
  cssClass?: string;
  name: string;
  title: string;
  type: 'radio' | 'checkbox';
  onToggleChange(): void;
}

export class ToggleSwitcher extends React.Component<IToggleSwitcherProps, {}> {
  constructor(props: IToggleSwitcherProps) {
    super(props);
  }

  public render() {
    return <div className={`c-toggle-switcher ${this.props.cssClass || ''}`}>
      <span className='c-toggle-switcher-title'>
        {this.props.title}
      </span>

      <div className='c-toggle-switcher-buttons btn-group btn-group-toggle' data-toggle='buttons'>
        {this.props.buttons.map((button, index) => this.getButton(button, index))}
      </div>
    </div>;
  }

  public getActiveButtonClass(type: string): string {
    return this.props.currentValue === type ? 'active' : '';
  }

  private getButton(button: IToggleButton, index: number) {
    return <label
    // tslint:disable-next-line:max-line-length
        className={`c-toggle-switcher-button btn ${this.props.buttonViewClass} ${button.value} ${this.getActiveButtonClass(button.value)}`}
        key={`c-toggle-switcher-button-${index}`}
    >
      <input
          type={this.props.type}
          name={this.props.name}
          value={button.value}
          onChange={this.props.onToggleChange}
          checked={this.props.currentValue === button.value}
      />
        {button.icon ? <i className={button.icon} /> : null}
        {button.label ? button.label : null}
    </label>;
  }
}
