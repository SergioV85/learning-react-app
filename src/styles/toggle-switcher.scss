@import './variables';

.c-toggle-switcher {
  &-title {
    color: $primary-dark;
    font-family: $main-font;
    font-size: 0.9rem;
    font-weight: 700;
  }
  &-button {
    color: $primary-dark;
    font-family: $main-font;
    font-size: 0.9rem;
    font-weight: 700;
    text-decoration: none;

    &:hover,
    &:active {
      text-decoration: none;
    }
  
    &.active {
      color: $light-red;
    }
  }
}