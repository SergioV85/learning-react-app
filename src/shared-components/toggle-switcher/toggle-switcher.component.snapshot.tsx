import { shallow } from 'enzyme';
import * as React from 'react';
import { ToggleSwitcher } from './toggle-switcher.component';

describe('Toggle Switcher Component Snapshot', () => {
  const mockedButtons = [
    {
      label: 'Rating',
      value: 'rating',
    },
    {
      label: 'Release Date',
      value: 'release_date',
    },
  ];
  const mockOnToggleChangeFn = jest.fn();

  it('should renders correctly', () => {
    const toggleSwitcherSnapshot = shallow(
      <ToggleSwitcher
        buttonViewClass='btn-link'
        buttons={mockedButtons}
        currentValue='rating'
        name='sort-by'
        title='Sort By'
        type='radio'
        onToggleChange={mockOnToggleChangeFn}
      />,
    );
    expect(toggleSwitcherSnapshot)
      .toMatchSnapshot();
  });
});
