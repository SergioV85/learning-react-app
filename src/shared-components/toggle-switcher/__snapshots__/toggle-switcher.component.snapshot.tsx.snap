// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Switcher Component Snapshot should renders correctly 1`] = `
<div
  className="c-toggle-switcher "
>
  <span
    className="c-toggle-switcher-title"
  >
    Sort By
  </span>
  <div
    className="c-toggle-switcher-buttons btn-group btn-group-toggle"
    data-toggle="buttons"
  >
    <label
      className="c-toggle-switcher-button btn btn-link rating active"
      key="c-toggle-switcher-button-0"
    >
      <input
        checked={true}
        name="sort-by"
        onChange={[MockFunction]}
        type="radio"
        value="rating"
      />
      Rating
    </label>
    <label
      className="c-toggle-switcher-button btn btn-link release_date "
      key="c-toggle-switcher-button-1"
    >
      <input
        checked={false}
        name="sort-by"
        onChange={[MockFunction]}
        type="radio"
        value="release_date"
      />
      Release Date
    </label>
  </div>
</div>
`;
